ui/src/pages/deployment-details/deployment-details-form/section-location.tsx

```diff
@@ -0,0 +1,123 @@
+import { FormField } from 'components/form/form-field'
+import { DeploymentFieldValues } from 'data-services/models/deployment'
+import { Button, ButtonTheme } from 'design-system/components/button/button'
+import { Map, MarkerPosition } from 'design-system/map/map'
+import _ from 'lodash'
+import { useContext, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { FormContext } from 'utils/formContext/formContext'
+import { isEmpty } from 'utils/isEmpty/isEmpty'
+import { STRING, translate } from 'utils/language'
+import styles from '../styles.module.scss'
+import { config } from './config'
+import { Section } from './deployment-details-form'
+
+type SectionLocationFieldValues = Pick<
+  DeploymentFieldValues,
+  'latitude' | 'longitude'
+>
+
+const DEFAULT_VALUES: SectionLocationFieldValues = {
+  latitude: 0,
+  longitude: 0,
+}
+
+export const SectionLocation = ({
+  onBack,
+  onNext,
+}: {
+  onBack: () => void
+  onNext: () => void
+}) => {
+  const {
+    formSectionRef,
+    formState,
+    setFormSectionStatus,
+    setFormSectionValues,
+  } = useContext(FormContext)
+
+  const defaultValues = {
+    ...DEFAULT_VALUES,
+    ..._.omitBy(formState[Section.Location].values, isEmpty),
+  }
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid },
+    setValue,
+  } = useForm<SectionLocationFieldValues>({
+    defaultValues,
+    mode: 'onBlur',
+  })
+
+  const [markerPosition, setMarkerPosition] = useState(
+    new MarkerPosition(defaultValues.latitude, defaultValues.longitude)
+  )
+
+  useEffect(() => {
+    setFormSectionStatus(Section.Location, { isDirty, isValid })
+  }, [isDirty, isValid])
```

Comment:
Maybe this could be extracted to a reusable hook? seems to be used in a couple of places.

Reply:
Good idea, done!